hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -594,15 +595,16 @@ def _networks(bits):
     _global_type_lookup[os._Environ] = st.just(os.environ)
 
 
-try:  # pragma: no cover
-    import numpy as np
-
-    from hypothesis.extra.numpy import array_dtypes, array_shapes, arrays, scalar_dtypes
-
-    _global_type_lookup[np.dtype] = array_dtypes()
-    _global_type_lookup[np.ndarray] = arrays(scalar_dtypes(), array_shapes(max_dims=2))
-except ImportError:
-    pass
+# These extras define a callable that either resolves to a strategy for this
+# narrowly extra-specific type, or returns None to proceed with normal type
+# resolution. The callable will only be called if the module is installed. To
+# avoid the performance hit of importing anything here, we defer it until the
+# method is actually called.
+_global_extra_lookup: typing.Dict[str, typing.Callable[[type], st.SearchStrategy]] = {
+    "numpy": lambda thing: importlib.import_module("hypothesis.extra.numpy")._from_type(
+        thing
+    ),
```

Comment:
I think I'd prefer a non-lambda function here - this is getting moderately complicated, and if it fails the traceback wouldn't be amazing.  At that point we can also do
```python
def _from_numpy_type(thing):
    from hypothesis.extra.numpy import _from_type

    _global_extra_lookup["numpy"] = _from_type
    return _from_type(thing)
```
for slightly faster / simpler subsequent calls.

Reply:
And in fact it could even remove itself from `_global_extra_lookup` on `ImportError`!